prevent-broker-scale-down-if-contains-partition-replicas.md

```diff
@@ -41,11 +30,12 @@ This class will also contain methods like `topicNames`, `describeTopics` to quer
 - Then we can use this information to check if the broker contains any partition replicas or not.
 - The scale down is done after we make sure that the brokers that are going to be removed doesn't contain any partition replicas. By doing this we avoid any partial scale down.
 
-#### What to do if a broker contains partitions?
+### What to do if a broker contains partitions?
 
 #### Flow:
 
-- If partition replicas are found out on the broker we will revert back the kafka replicas to the previous count by changing the replica count directly in the kafka object and update the status with a warning stating that the scale down is not done and the replica count is changed back to previous one in the memory.
+- If partition replicas are found out on the broker we will revert back the kafka replicas to the previous count by changing the `spec.replicas` in the STS/SPS and the rest of the reconciliation will be done normally.
+- We will also generate a new condition which will be added to Kafka resource status depicting that the scale down is not done. It will also contain the `spec.replicas` count(which is being currently being used) in the condition message.
```

Comment:
Can you give an example YAML?

Would the fact that this resource wasn't completely reconciled be apparent from any metrics? Off the top of my head this would be the first time a resource with a `Ready` condition wasn't completely reconciled. I'm not sure how well that aligns with user expectations, TBH. We might need to add a metric so that users can alert on resources like this.

Reply:
@tombentley Hi, do you want the yaml with the warning condition?

Reply:
This is the Kafka resource yaml when the scaled down fails --> https://gist.github.com/ShubhamRwt/1d26531f5866529eba85820b93825110. I have added the `status` section to the proposal also. Iam not sure on if we need metrics or not for this since we are not failing the reconciliation. @ppatierno @scholzj wdyt?

Reply:
Well, until we have some auto-scaling solution, they will probably notice it:
* They will take a manual action and expect to see the result
* It will not be just the status warning they will see, they will also see the pods which did not scale down

So I think this will be more of a problem if we ever have some form of auto-scaling.

Reply:
Good point from Jakub. User has ways to see that what they asked for didn't happen. Anyway, I am not against having a metric. It's just not sure how much it's useful today.